cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/CommandUtil.java

```diff
@@ -207,6 +247,364 @@ public static void applyTemplate(Path modulePath, String template) throws IOExce
         }
     }
 
+    /**
+     * Apply the central template to the created module.
+     *
+     * @param modulePath path to the module
+     * @param balaCache path to balacache
+     * @param template template name
+     */
+    public static void applyBalaTemplate(Path modulePath, Path balaCache, String template) {
+        // find all balas matching org and package name.
+        Path balaTemplate = (Path) findBalaTemplate(template);
+        String packageName = findPkgName(template);
+        String orgName = findOrg(template);
+        String version = findPkgVersion(template);
+
+        if (version == null) {
+            Path balaToPkgPath = balaCache.resolve(orgName).resolve(packageName);
+            List<PackageVersion> packageVersions = getPackageVersions(balaToPkgPath);
+
+            if (packageVersions.size() == 1) {
+                version = String.valueOf(packageVersions.get(0));
+            } else {
+                version = String.valueOf(findLatest(packageVersions));
+            }
+        }
+
+        Path balaPath = balaCache.resolve(
+                ProjectUtils.getRelativeBalaPath(orgName, packageName, version, null));
+        //First we will check for a bala that match any platform
+        String platform = findPlatform(balaPath);
+        balaPath = balaCache.resolve(
+                ProjectUtils.getRelativeBalaPath(orgName, packageName, version, platform));
+
+        Gson gson = new Gson();
+        Path packageJsonPath = balaTemplate.resolve("package.json");
+        if (balaTemplate != null) {
+            try (FileReader packageReader = new FileReader(String.valueOf(packageJsonPath))) {
+                PackageJson packageJson = gson.fromJson(packageReader, PackageJson.class);
+                boolean templateVal = packageJson.getTemplate();
+                if (templateVal) {
```

Comment:
A boolean variable `templateVal` is a bit redundant if its only being used once. Maybe have `if (packageJson.getTemplate())` instead? 

Same applies to line `558`